jsonrpc/codec.go

```diff
@@ -169,14 +169,31 @@ func (bnh *BlockNumberOrHash) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// stringToBlockNumberSafe parses string and returns Block Number
+// works similar to stringToBlockNumber
+// but treats empty string or pending block number as a latest
+func stringToBlockNumberSafe(str string) (BlockNumber, error) {
+	switch str {
+	case "", pending:
+		return LatestBlockNumber, nil
+	default:
+		return stringToBlockNumber(str)
+	}
+}
```

Comment:
WDYT?

```suggestion
func stringToBlockNumberSafe(str string) BlockNumber {
	switch str {
	case "", pending:
		return LatestBlockNumber
	default:
		blockNum, _ := stringToBlockNumber(str)
		
		return blockNum
	}
}
```

Reply:
this is not good, 
```
	case latest:
		return LatestBlockNumber, nil
	case earliest:
		return EarliestBlockNumber, nil
```
should be covered also, that's why i am calling original func

Reply:
You have overlooked my proposed changes, I'm just suggesting to discard the error and remove the error from return values, since `stringToBlockNumber` returns an error just in case empty string was provided (which here wouldn't be the case, because it is handled in the `case` statement before) :slightly_smiling_face:

Reply:
My bad, we cannot make the suggested change, since `stringToBlockNumber` can return an error in case a string which is not parseable to `uint64` is provided.